app/scripts/lib/ppom/ppom-middleware.ts

```diff
@@ -78,7 +79,7 @@ export function createPPOMMiddleware<
       const { chainId } =
         networkController.getNetworkConfigurationByNetworkClientId(
           req.networkClientId,
-        )!;
+        ) ?? {};
```

Comment:
Why is there a fallback? If we can't get the chain ID, we need to throw. It would also remove the need for the cast `as Hex` below.

Doesn't `getNetworkConfigurationByNetworkClientId` already throw if it doesn't recognise the `networkClientId`?

Reply:
This is as TS is complaining about destructuring for possible undefined value.

Reply:
It looks like that method in the `NetworkController` actually just reads a map, so do we need a throw if there is no result found, rather than an empty object and cast?

https://github.com/MetaMask/core/blob/main/packages/network-controller/src/NetworkController.ts#L1988

Reply:
I added this code below above, but the casting is needed for linting for pass

```
      if (!chainId) {
        throw Error('ChainID not found for networkClientId');
      }
```